frontend/src/components/markdown/DiagramEditorDialog.tsx

```
@@ -0,0 +1,367 @@
+          {/* Right side - Preview and Templates */}
+          <div className="flex-1 flex flex-col gap-4">
+            <div className="flex-1">
+              <Label>Preview</Label>
+              <Card className="h-64 mt-2 overflow-auto">
+                <div
+                  ref={previewRef}
+                  className="p-4 flex items-center justify-center min-h-full"
+                >
+                  {renderedDiagram ? (
+                    <div
+                      dangerouslySetInnerHTML={{ __html: renderedDiagram }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)